### PR TITLE
Add Dockerfile, environment.yml and dependencies to setup.py

### PR DESCRIPTION
**Pull request by jorgensd**
- Create a Conda environment file for TurtleFSI.
- Add dockerfile building on the conda package
- Change CI env to the `scientific-computing-env`, running DOLFIN on a ubuntu 22.04 system with Python 3.10 (main branch as of 8th of December).

### CODE REVIEW

**Comment by jorgensd** [comment]
@johannesring This is now ready for review. Do you have anything to add to this?:)

**Comment by jorgensd** [comment]
A potential further improvement would be to add a TurtleFSI package to Github, similar to: https://github.com/scientificcomputing/example-paper-fenics/pkgs/container/example-paper-fenics
which would amount to adding: https://github.com/scientificcomputing/example-paper-fenics/blob/main/.github/workflows/docker-image.yml
Is that something that anyone would be interested in having?
Then we could run a pre-installed TurtleFSI docker image as 
`docker run -ti ghcr.io/KVSlab/turtleFSI:vx.y.y`

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  - conda-forge
+dependencies:
+  - fenics-dolfin
+  - python=3.10
+  - pip
+  - git
+  - scipy
+  - pip:
+    - cppimport
+    - git+https://github.com/mikaem/fenicstools
```
Comment (johannesring) [inline environment.yml:13]: Maybe remove `fenicstools` since it is not used by `turtleFSI`?

**Review by johannesring — APPROVED** [review]
Looks good!

File: environment.yml (hunk 1 above)
Comment (jorgensd) [inline environment.yml:13]:
```suggestion

```

**Pull request merged**